tests/toranj/build.sh

```diff
@@ -174,86 +168,6 @@ case ${build_config} in
         cp -p ${top_builddir}/examples/apps/ncp/ot-ncp-ftd ${top_builddir}/examples/apps/ncp/ot-ncp-ftd-15.4-trel
         ;;
 
-    rcp)
-        echo "===================================================================================================="
-        echo "Building OpenThread RCP (NCP in radio mode) with simulation platform"
-        echo "===================================================================================================="
-        ./bootstrap || die "bootstrap failed"
-        cd "${top_builddir}" || die "cd failed"
-        cppflags_config='-DOPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"../tests/toranj/openthread-core-toranj-config-simulation.h\"'
-        ${top_srcdir}/configure \
-            CPPFLAGS="$cppflags_config " \
-            --enable-coverage=${coverage} \
-            --enable-ncp \
-            --enable-radio-only \
-            --with-examples=simulation \
-            --disable-docs \
-            --enable-tests=$tests || die
-        make -j 8 || die
-        ;;
-
-    posix | posix-)
```

Comment:
Actually I suggest keeping all the rest of configurations here in the `build.sh`. There is no harm/overhead in keeping them (even if they may not be by any CI).

It's useful tool to quickly configure a build flavor and test builds during development.

Reply:
After removing the NCP app, it seems these options cannot build valid binaries for toranj tests. Are you sure want to keep these options?

Reply:
Yes. Let's keep them. We can change them to build CLI instead of NCP.

Reply:
Reverted changes in build.sh.

Reply:
I think we may need to change the script for this build flavors to actually build CLI (and not NCP anymore).
It's ok I can take care of later in separate PR.